addons/mail/static/src/models/thread/thread.js

```diff
@@ -2022,6 +2022,7 @@ function factory(dependencies) {
             compute: '_onChangeThreadViews',
             dependencies: [
                 'threadViews',
+                'needactionMessagesAsOriginThread',
```

Comment:
Alphabetical order.

Btw adding this dependency means if we receive a new notification for a thread already opened, the notification will instantly disappear. I don't like it but I don't like this whole feature either so I guess that's ok...

Reply:
I just tested and confirm that the notification is directly marked as read.
I don't think it's ok actually.
My guess is that people have multiple thread open across multiple tabs and all those thread will not trigger any notifications at all.

One way to go around this is to go back in the very first version I made and trigger `markNeedactionMessagesAsOriginThreadAsRead` inside the mounted method of the chatter component. That way, only the first load of the chatter will mark as  read.
I don't really remember why that version was wrong, probably just because we don't like doing stuff in components ?

Reply:
We will still have the same problem we are trying to fix if we move to mounted, because I'm guessing the issue is that when opening in a new tab the thread itself is shown before the inbox is loaded (which, when inbox is loaded, is considered the same way as if a new notification was received).

Reply:
With https://github.com/odoo/odoo/pull/64009 we don't use the message id's anymore, the need actions are marked as read regardless of the inbox.

Reply:
Ok that's good, I guess then current PR can be closed. Make sure the multi-tab is fixed with the other PR.